libraries/botbuilder-middleware-engine-core/src/engine.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * @module botbuildercommunity/middleware-engine-core
+ */
+
+export abstract class Engine {
+    public client: any;
+    public async abstract entities(input: any): Promise<any>;
+    public async abstract keyPhrases(input: any): Promise<any>;
+    public async abstract detectLanguage(input: any): Promise<any>;
+    public async abstract sentiment(input: any): Promise<any>;
+    public async abstract categories(input: any): Promise<any>;
+    public async abstract concepts(input: any): Promise<any>;
+    public async abstract emotion(input: any): Promise<any>;
```

Comment:
My assumption is that the base engine with change if other text analysis is added. It usually takes three rounds of additions and abstractions before getting the outer framework correct. I purposely put this in here as a reminder for further abstractions and/or possible custom implementations (i.e., creating emotion detection via CNTK to enhance Cognitive Services). I actually have exception messages in the CS and Watson engines if you call one of these that aren't implemented. My home to for these engines to seem identical and swappable, especially since the middleware class names are identical. If AWS, Google, or Facebook text analysis is added in the future, most will have similar text processing (i.e., all will likely have sentiment analysis), so my goal is to identify _all_ text analysis, and then have "not supported" exception.

We'll see if it works out that way...